code/modules/mob/living/carbon/human/species_types/zombies.dm

```diff
@@ -46,15 +47,25 @@
 /datum/species/zombie/infectious/spec_life(mob/living/carbon/C)
 	. = ..()
 	C.a_intent = INTENT_HARM // THE SUFFERING MUST FLOW
+	
+	//Zombies never actually die, they just fall down until they regenerate enough to rise back up.
+	//They must be restrained, beheaded or gibbed to stop being a threat.
 	if(regen_cooldown < world.time)
-		C.heal_overall_damage(4,4)
-		C.adjustToxLoss(-4)
-	if(prob(4))
+		var/heal_amt = heal_rate
+		if(C.InCritical())
+			heal_amt *= 2
+		C.heal_overall_damage(heal_amt,heal_amt)
+		C.adjustToxLoss(-heal_amt)
+	if(!C.InCritical() && prob(4))
 		playsound(C, pick(spooks), 50, TRUE, 10)
-	if(C.InCritical())
-		C.death()
-		// Zombies only move around when not in crit, they instantly
-		// succumb otherwise, and will standup again soon
+		
+//Congrats you somehow died so hard you stopped being a zombie
+/datum/species/zombie/infectious/spec_death(mob/living/carbon/C)
+	. = ..()
+	var/obj/item/organ/zombie_infection/infection
+	infection = C.getorganslot(ORGAN_SLOT_ZOMBIE)
+	if(infection)
+		qdel(infection)

```

Comment:
QDEL_NULL probably

Reply:
it's a local var tho, does it make sense?

Reply:
nvm actually